src/groupdef.cpp

```diff
@@ -327,13 +327,16 @@ bool GroupDefImpl::addNamespace(const NamespaceDef *def)
   return FALSE;
 }
 
-void GroupDefImpl::addDir(const DirDef *def)
+void GroupDefImpl::addDir(DirDef *const def)
 {
   if (def->isHidden()) return;
   if (Config_getBool(SORT_BRIEF_DOCS))
-    m_dirList->inSort(def);
+  {
+    m_dirList->push_back(def);
+    m_dirList->sort();
```

Comment:
Does this really sort the objects, or the items?

Reply:
`QList::inSort()` [is using](https://github.com/doxygen/doxygen/blob/91370bf84ac299fcb773e1b9d81e8f5c56da0725/qtools/qlist.doc#L223) `compareItems()`.

Reply:
`inSort()` inserts one element at the right place, whereas `sort()` sorts the whole list. In most places I use append (push_back) while adding and then sort once after everything is in the list. so `sort()` is not a good replacement for `inSort()`

Reply:
I copied the QList way of `inSort` into [a free function](https://github.com/doxygen/doxygen/pull/7728/files#diff-c0b61dd85e8f538c4e4e369f0e52cb4cR1101-R1109).